File: ruby/lasagna/lasagna.rb
class Lasagna
  EXPECTED_MINUTES_IN_OVEN = 40

  def remaining_minutes_in_oven(actual_minutes_in_oven)
    if actual_minutes_in_oven > Lasagna::EXPECTED_MINUTES_IN_OVEN
      0
    else
      Lasagna::EXPECTED_MINUTES_IN_OVEN - actual_minutes_in_oven
    end
  end

  def preparation_time_in_minutes(layers)
    layers * preparation_time_per_layer    
  end

  def total_time_in_minutes(number_of_layers:, actual_minutes_in_oven:)
    preparation_time_in_minutes(number_of_layers) + actual_minutes_in_oven
  end

  private

  def preparation_time_per_layer
    2
  end
end

File: lua/lasagna/lasagna.lua
local lasagna = {
	oven_time = 40,
}

-- add a 'oven_time' field to the lasagna table

-- returns the remaining minutes based on the actual minutes in the oven
function lasagna.remaining_oven_time(actual_minutes_in_oven)
	return lasagna.oven_time - actual_minutes_in_oven
end

-- calculates and returns the time needed to prepare the lasagna on the amount
-- of layers
function lasagna.preparation_time(number_of_layers)
	return number_of_layers * 2
end

-- calculates the time elapsed cooking the lasagna - including preparation time
-- and baking time.
function lasagna.elapsed_time(number_of_layers, actual_minutes_in_oven)
	return lasagna.preparation_time(number_of_layers) + actual_minutes_in_oven
end

return lasagna
